cmd/controller/main.go

```diff
@@ -31,17 +32,18 @@ import (
 )
 
 type flags struct {
-	extProcLogLevel        string
-	extProcImage           string
-	extProcImagePullPolicy corev1.PullPolicy
-	enableLeaderElection   bool
-	logLevel               zapcore.Level
-	extensionServerPort    string
-	tlsCertDir             string
-	tlsCertName            string
-	tlsKeyName             string
-	caBundleName           string
-	envoyGatewayNamespace  string
+	extProcLogLevel                  string
+	extProcImage                     string
+	extProcImagePullPolicy           corev1.PullPolicy
+	enableLeaderElection             bool
+	logLevel                         zapcore.Level
+	extensionServerPort              string
+	tlsCertDir                       string
+	tlsCertName                      string
+	tlsKeyName                       string
+	caBundleName                     string
+	envoyGatewayNamespace            string
+	metricsRequestHeaderLabelMapping string
```

Comment:
could you store map here already since you are unnecessarily calling `parseRequestHeaderLabelMapping` twice due to this?
```suggestion
	metricsRequestHeaderLabelMapping map[string]string
```

Reply:
ah now i see that two `parseRequestHeaderLabelMapping` exists with exactly the same logic. Could you put `parseRequestHeaderLabelMapping` in `internalapi` and share the code?

Reply:
and using `string` makes sense here on the other hand. Thank you for explaining it